src/pkg/runner/actions_test.go

```diff
@@ -426,3 +427,93 @@ func TestRunner_processAction(t *testing.T) {
 		})
 	}
 }
+
+func TestRunner_GetBaseActionCfg(t *testing.T) {
+	type args struct {
+		cfg      types.ActionDefaults
+		a        types.BaseAction[string]
+		vars     variables.SetVariableMap[string]
+		extraEnv map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "ActionDefaults used when no overrides",
+			args: args{
+				cfg: types.ActionDefaults{
+					Env: []string{"ENV1=fromDefault", "ENV2=xyz"},
+				},
+				a: types.BaseAction[string]{},
+				//vars: variables.SetVariableMap[string]{"ENV1": {Value: "fromSet"}},
+				//extraEnv: map[string]string{"ENV1": "fromExtra"},
+			},
+			want: []string{"ENV1=fromDefault", "ENV2=xyz"},
+		},
+		{
+			name: "extraEnv overrides defaults",
+			args: args{
+				cfg: types.ActionDefaults{
+					Env: []string{"ENV1=fromDefault", "ENV2=xyz1"},
+				},
+				a:        types.BaseAction[string]{},
+				vars:     variables.SetVariableMap[string]{"ENV1": {Value: "fromSet"}},
+				extraEnv: map[string]string{"ENV1": "fromExtra"},
+			},
+			want: []string{"ENV1=fromExtra", "ENV2=xyz1"},
```

Comment:
is this right for this test? (would expect the extra to be at the end with 4 total elements)

Reply:
Oh, I need to update the want logic here to check for exact match with the array. It currently only checks that the want ones are present and ignores the rest. Fixing.